presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileFunctions.java

```diff
@@ -234,13 +234,14 @@ private static BingTile getTileCoveringLowerRightCorner(Envelope envelope, int z
         return tile;
     }
 
-    private static void checkGeometryToBingTilesLimits(OGCGeometry geometry, boolean pointOrRectangle, int tileCount)
+    private static void checkGeometryToBingTilesLimits(OGCGeometry geometry, boolean pointOrRectangle, long tileCount)
     {
         if (pointOrRectangle) {
             checkCondition(tileCount <= 1_000_000, "The number of input tiles is too large (more than 1M) to compute a set of covering bing tiles.");
         }
         else {
-            long complexity = ((long) tileCount) * getPointCount(geometry.getEsriGeometry());
+            checkCondition((int) tileCount == tileCount, "The zoom level is too high to compute a set of covering bing tiles.");
```

Comment:
Why is this check needed? Doesn’t the check below cover it? In fact, for complexity to be < 25M, tilecount cannot be larger than that.

Reply:
Also, on L183 you call `toIntExact(tileCount)`, which will throw in this case anyway.

Reply:
> Why is this check needed?

@martint This check is needed to avoid long overflow when calculating `complexity`.

@nezihyigitbasi  `toIntExact(tileCount)` throws `ArithmeticException`, which is not helpful to the user. With explicit check I can provide a clear error message.

Reply:
@mbasmanova That check may not catch long overflow when calculating complexity. Assume that `tileCount` is `Integer.MAX_VALUE` and the number of points is, say, 306783378, then `complexity = Integer.MAX_VALUE * 306783378;`, which will be `-306783378` (overflow).

Reply:
@nezihyigitbasi What's the best way to address that? Should I use `Math.multiplyExact`, catch `ArithmeticException` and translate it to `PrestoException(INVALID_FUNCTION_ARGUMENT, ...)`?

Reply:
That sounds reasonable.